tests/test_banding.py

```diff
@@ -0,0 +1,105 @@
+#       of its contributors may be used to endorse or promote products
+#       derived from this software without specific prior written
+#       permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+# Contact: khmer-project@idyll.org
+# pylint: disable=missing-docstring,protected-access,no-member,invalid-name
+from __future__ import print_function
+from __future__ import absolute_import, division
+
+import screed
+import khmer
+from . import khmer_tst_utils as utils
+import pytest
+
+
+@pytest.mark.parametrize('ksize,memory,epsilon,numbands', [
+    (21, 5e7, 1, 2),
+    (21, 5e7, 1, 4),
+    (21, 5e7, 1, 8),
+    (21, 5e7, 2, 16),
+])
+def test_banding_in_memory(ksize, memory, epsilon, numbands):
+    infile = utils.get_test_data('test-reads.fa')
+
+    ct_normal = khmer.Counttable(ksize, memory / 4, 4)
+    ct_normal.consume_fasta(infile)
+
+    ct_banded = list()
+    for band in range(numbands):
+        ct = khmer.Counttable(ksize, memory / 4 / numbands, 4)
+        ct.consume_fasta_banding(infile, numbands, band)
+        ct_banded.append(ct)
+
+    for n, record in enumerate(screed.open(infile)):
+        if not (n > 0 and n % 100 == 0):
+            continue
+        for kmer in ct_normal.get_kmers(record.sequence):
+            abund_normal = ct_normal.get(kmer)
+            abunds_banded = [ct.get(kmer) for ct in ct_banded]
+            # Ideally, we'd like to enforce that these are equal. Practically,
+            # with false positives, we have to allow for a small difference.
+            assert abs(sum(abunds_banded) - abund_normal) <= epsilon
+
+            nonzeros = [a for a in abunds_banded if a > 0]
+            # False positives shouldn't be appearing in multiple bands
+            assert len(nonzeros) <= 2
+            # False positives shouldn't have high abundance
+            if len(nonzeros) > 1:
+                assert min(nonzeros) == 1
+
+
+@pytest.mark.parametrize('ksize,memory,numbands', [
+    (21, 5e7, 2),
+    (21, 5e7, 4),
+    (21, 5e7, 8),
+    (21, 5e7, 16),
+])
+def test_banding_to_disk(ksize, memory, numbands):
+    infile = utils.get_test_data('test-reads.fa')
+    path1 = utils.get_temp_filename('normal.ct')
+    path2 = utils.get_temp_filename('banding.ct')
```

Comment:
This function is simple enough, and it provides a stronger guarantee. Perhaps it's all we need?